arch/x86/include/intel64/kernel_arch_data.h

```diff
@@ -26,6 +26,7 @@ struct x86_cpuboot {
 	size_t stack_size;	/* size of stack */
 	arch_cpustart_t fn;	/* kernel entry function */
 	void *arg;		/* argument for above function */
+	uint8_t cpu_id;		/* APIC ID */
```

Comment:
Shouldn't this be guarded under `CONFIG_ACPI`?

Reply:
In that case, it would be a good idea to rename it to `apic_cpu_id` or something similar. The name `cpu_id` is used in subsequent commits where it is Zephyr's CPU ID and not APIC. Renaming it will avoid any confusion between two.

Reply:
sorry my mistake, this is cpu id pass down by kernel (not the HW CPU ID which is same as local APIC ID). Updated the comment